sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2Suite.scala

```diff
@@ -70,6 +73,34 @@ class HiveThriftServer2Suite extends FunSuite with Logging {
     port
   }
```

Comment:
`randomListeningPort` needs to be replaced by something like this:

```scala
  def randomListeningPorts(n: Int) = {
    val sockets = Array.fill(n)(new ServerSocket(0))
    val ports = sockets.map(_.getLocalPort)
    sockets.foreach(_.close())
    ports
  }
```

Because now we need to assign two ports (server and UI).